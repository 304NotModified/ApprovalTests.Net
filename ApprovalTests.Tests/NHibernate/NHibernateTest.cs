using System.Linq;
using System.Reflection;
using ApprovalTests.Persistence.NHibernate;
using NHibernate;
using NHibernate.Cfg;
using NHibernate.Linq;
using NUnit.Framework;

namespace ApprovalTests.Tests.NHibernate
{
    using System;

    [TestFixture]
    public class NHibernateTest
    {
        public static ISessionFactory SessionFactory;

        public static ISession OpenSession()
        {
            if (SessionFactory == null) //not threadsafe
            {
                //SessionFactories are expensive, create only once
                Configuration configuration = new Configuration();
                configuration.AddAssembly(Assembly.GetCallingAssembly());
                SessionFactory = configuration.BuildSessionFactory();
            }
            return SessionFactory.OpenSession();
        }

        [Test]
        public void TestSimpleQuery()
        {
            if (Environment.MachineName != "LLewellyn's machine")
            {
                Assert.Ignore("Only works on Llewellyn's machine.");
            }

            using (ISession session = OpenSession())
            {
                IQueryable<Company> query =
                    from a in session.Query<Company>()
                    where a.Name.StartsWith("Mic")
                    select a;
                NHibernateApprovals.Verify((NhQueryable<Company>)query);
            }
        }
    }
}