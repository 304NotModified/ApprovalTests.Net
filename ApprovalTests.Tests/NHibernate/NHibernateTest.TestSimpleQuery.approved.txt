
    select
        company0_.Id as Id0_,
        company0_.Name as Name0_,
        company0_.Website as Website0_ 
    from
        Company company0_ 
    where
        company0_.Name like ('Mic'+'%')